backend/server/websocket_manager.py

```
@@ -61,43 +62,53 @@ async def connect(self, websocket: WebSocket):
+                except Exception as e:
+                    logger.info(f"WebSocket already closed: {e}")
+        except Exception as e:
+            logger.error(f"Error during WebSocket disconnection: {e}")
+            # Still try to close the connection if possible
             try:
                 await websocket.close()
             except:
-                pass  # Connection might already be closed
+                pass  # If this fails too, there's nothing more we can do
 
     async def start_streaming(self, task, report_type, report_source, source_urls, document_urls, tone, websocket, headers=None, query_domains=[], mcp_enabled=False, mcp_strategy="fast", mcp_configs=[]):
```

Ruff (0.14.3)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)